lib/ExpensiMark.ts

```diff
@@ -260,23 +324,43 @@ export default class ExpensiMark {
-                    const replaceFunction = (g1) => replacement(g1, shouldKeepRawInput);
                     if (shouldKeepRawInput) {
                         const rawInputRegex = /^(?:&gt;)+ +(?! )(?![^<]*(?:<\/pre>|<\/code>))([^\v\n\r]*)/gm;
-                        return textToProcess.replace(rawInputRegex, replaceFunction);
+                        if (typeof replacement === 'function') {
+                            return textToProcess.replace(rawInputRegex, (...args) => replacement(EXTRAS_DEFAULT, ...args));
+                        }
+                        return textToProcess.replace(rawInputRegex, replacement);
                     }
-                    return this.modifyTextForQuote(regex, textToProcess, replacement);
+                    return this.modifyTextForQuote(regex, textToProcess, replacement as ReplacementFn);
+                },
+                replacement: (_extras, g1) => {
+                    // We want to enable 2 options of nested heading inside the blockquote: "># heading" and "> # heading".
+                    // To do this we need to parse body of the quote without first space
+                    const handleMatch = (match: string) => match;
+                    const textToReplace = g1.replace(/^&gt;( )?/gm, handleMatch);
+                    const filterRules = ['heading1'];
+
+                    // if we don't reach the max quote depth we allow the recursive call to process possible quote
+                    if (this.currentQuoteDepth < this.maxQuoteDepth - 1) {
+                        filterRules.push('quote');
+                        this.currentQuoteDepth++;
+                    }
+
+                    const replacedText = this.replace(textToReplace, {
+                        filterRules,
+                        shouldEscapeText: false,
+                        shouldKeepRawInput: false,
+                    });
+                    this.currentQuoteDepth = 0;
+                    return `<blockquote>${replacedText}</blockquote>`;
                 },
-                replacement: (g1, shouldKeepRawInput = false) => {
+                rawInputReplacement: (_extras, g1) => {
                     // We want to enable 2 options of nested heading inside the blockquote: "># heading" and "> # heading".
                     // To do this we need to parse body of the quote without first space
                     let isStartingWithSpace = false;
-                    const handleMatch = (match, g2) => {
-                        if (shouldKeepRawInput) {
-                            isStartingWithSpace = !!g2;
-                            return '';
-                        }
-                        return match;
+                    const handleMatch = (_match: string, g2: string) => {
+                        isStartingWithSpace = !!g2;
+                        return '';
                     };
                     const textToReplace = g1.replace(/^&gt;( )?/gm, handleMatch);
```

Comment:
Why now we aren't returning `match` in handleMatch function if `isStartingWithSpace` is false?

Reply:
Previously there was no `rawInputReplacement` function. What I did is to split previous logic into two functions (`replacement` and `rawInputReplacement`). `match` is returned but only in `replacement` function above